### PR TITLE
In Community Sidebar before Chain

### PR DESCRIPTION
**Pull request by dillchen**
Following @mzparacha's PR #4182, this potentially allows the "community sidebar" to load before requiring the whole chain to be loaded. This does feel smoother. See Looms at the bottom.

## Link to Issue
Closes: #4196 

## Description of Changes
- Fixed a UI glitch when using sidebar button options in communities
- Fixed a blank page suspense bug, added a proper loading indicator
- Removed extra boilerplate
- Removed <Sublayout/> from pages and added it to layout. Now we have 3 layout types, `blank` which is just a blank page with some layout css, `common` which is our app specific - pages that are specifically owned by commonwealth like settings, user profile (its just b/w the user and cw - no communities or third party involved) - it has the commonwealth sidebar (the leftmost sidebar) and header, `community` is the last layout type which has community sidebar along with commonwealth sidebar and header. `community` is the default layout type - i set it to default because it was the most used layout type and i didnt want to drill the prop from higher comps here.
- Community sidebar now renders before the data is loaded, we decoupled it from the layout menu type and now it works based on `layout` type, if `community` is the layout, it would render before without looking for `menuName` which is set in zustand.
- Added a splash screen with a still (non-moving) cow bobber. It will only be displayed once on site load.

## Test Plan
We will have to test each page, since the pr changes some layout code which is basically the entry point for app.
- Components rendered in community pages should have the `header`, `commonwealth sidebar` and the `community sidebar`
- Components rendered in commonwealth specific pages should have the `header` and `commonwealth sidebar` 
- The landing page should have the `blank` layout, i.e no header or sidebars.
- Switching b/w pages should change layout properly
- Suspense fallback should be triggered properly when visiting pages that suspend like the /overview page of communities
- Clicking on sidebar buttons should not cause a layout glitch in communities
- The splash screen with a non-moving cow bobber should be visible on page load

## Deployment Plan
N/A

## Other Considerations
N/A

### Before
https://www.loom.com/share/8b3de254b6034e23890277aa08baa45c

### After
https://www.loom.com/share/fee52d446c2c414aa49bf7214f7bf5e1

### CODE REVIEW

**Comment by mzparacha** [comment]
Not sure if this is the correct time to review, maybe its better to review after [this](https://github.com/hicommonwealth/commonwealth/issues/4196#issuecomment-1587885178) change?

**Comment by jnaviask** [comment]
@mzparacha my expectation is your PR to fix #4196 will supersede this PR, although the changes here may be useful for guiding your work.

**Comment by jnaviask** [comment]
@mzparacha if you're planning to take over this PR, please mark as Draft until Ready for Review.

**Comment by mzparacha** [comment]
@dillchen @CowMuon @jnaviask Ready for review!

The layout should feel a bit less glitchy now, but please lmk if you find any issues.

**Comment by mzparacha** [comment]
Also to point out we can improve the search bar and right side profile menu in header to be a bit less flanky/glitchy, specially the search bar. Do we want to include it in this pr?

File: packages/commonwealth/client/scripts/App.tsx [hunk 1]
```diff
@@ -6,14 +6,28 @@ import React, { StrictMode } from 'react';
 import { RouterProvider } from 'react-router-dom';
 import { ToastContainer } from 'react-toastify';
 import { queryClient } from 'state/api/config';
+import { CWIcon } from './views/components/component_kit/cw_icons/cw_icon';
+
+const Splash = () => {
+  return (
+    <div className="Splash">
```
Comment (mzparacha) [inline packages/commonwealth/client/scripts/App.tsx:13]: Splash ui

File: packages/commonwealth/client/scripts/views/Layout.tsx [hunk 2]
```diff
@@ -135,58 +98,68 @@ const LayoutComponent = ({
...
+      </Suspense>
+    );
+  };
 
-export const withLayout = (Component, params) => {
   return (
     <ErrorBoundary
       FallbackComponent={({ error }) => <ErrorPage message={error?.message} />}
     >
-      <Suspense fallback={null}>
```
Comment (mzparacha) [inline packages/commonwealth/client/scripts/views/Layout.tsx:187]: the `null` here caused the blank screen for a brief time

File: packages/commonwealth/client/scripts/views/components/sidebar/index.tsx [hunk 3]
```diff
@@ -47,47 +45,43 @@ export const Sidebar = () => {
...
-              {showAdmin && <AdminSection />}
-              {featureFlags.communityHomepage && app.chain.meta.hasHomepage && (
-                <div
-                  className={onHomeRoute ? 'home-button active' : 'home-button'}
-                  onClick={() => navigate('/feed')}
-                >
-                  <CWIcon iconName="home" iconSize="small" />
-                  <CWText>Home</CWText>
+      <div className="sidebar-default-menu">
+        <SidebarQuickSwitcher />
```
Comment (mzparacha) [inline packages/commonwealth/client/scripts/views/components/sidebar/index.tsx:49]: the commonwealth sidebar will now not cause the ui glitch when clicked in `community` layout pages

**Comment by jnaviask** [comment]
Feels amazing. Just one small issue: sidebar state is weird for new snapshot proposal and new on-chain proposal. It should collapse but instead it pops back out after you click it.

**Comment by mzparacha** [comment]
ok so this bug was present before but the reload layout workaround nuked it, fixed in https://github.com/hicommonwealth/commonwealth/pull/4195/commits/73e28fe2a9dc64ddce185dc509eec52e9e4c0182

**Review by jnaviask — APPROVED** [review]

**Pull request merged**